crates/ruff_server/src/server/api/requests/code_action.rs

```diff
@@ -1,81 +1,129 @@
-use crate::edit::ToRangeExt;
+use crate::lint::fixes_for_diagnostics;
 use crate::server::api::LSPResult;
+use crate::server::SupportedCodeActionKind;
 use crate::server::{client::Notifier, Result};
 use crate::session::DocumentSnapshot;
+use crate::DIAGNOSTIC_NAME;
+use lsp_server::ErrorCode;
 use lsp_types::{self as types, request as req};
-use ruff_text_size::Ranged;
+use rustc_hash::FxHashSet;
+use types::{CodeActionKind, CodeActionOrCommand};
 
-pub(crate) struct CodeAction;
+use super::code_action_resolve::resolve_edit_for_fix_all;
 
-impl super::RequestHandler for CodeAction {
+pub(crate) struct CodeActions;
```

Comment:
Nit: It might be helpful to document the different code actions. What are they supposed to do and how do they work with other code actions.